From 59deb0d9325cb3c10202eaebfc6ff3aa915a1e34 Mon Sep 17 00:00:00 2001
From: Tim Kryger <tkryger@google.com>
Date: Fri, 28 Jul 2017 14:57:24 -0700
Subject: [PATCH 3/3] Reading the serialno property is forbidden

This read always fails so just remove it to avoid selinux denials.

Bug: 63963344
Test: Ran downloader utility on Angelfish over WIFI
Change-Id: I6ce7bb5d161c9d2c4478b5d0fa23dd64b6d599b3
Merged-In: I6ce7bb5d161c9d2c4478b5d0fa23dd64b6d599b3
(cherry picked from commit 1744da2bed9c6c41188b224da77d0e372d562eed)

# Conflicts:
#	wcnss-service/wcnss_service.c
---
 wcnss-service/wcnss_service.c | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/wcnss-service/wcnss_service.c b/wcnss-service/wcnss_service.c
index eafa81b..b67772d 100644
--- a/wcnss-service/wcnss_service.c
+++ b/wcnss-service/wcnss_service.c
@@ -406,10 +406,8 @@ void setup_wcnss_parameters(int *cal)
 #endif
 {
 	char msg[WCNSS_MAX_CMD_LEN];
-	char serial[PROPERTY_VALUE_MAX];
 	int fd, rc, pos = 0;
 	struct stat st;
-	unsigned int serial_num = 0;
 
 	fd = open(WCNSS_CTRL, O_WRONLY);
 	if (fd < 0) {
@@ -417,25 +415,6 @@ void setup_wcnss_parameters(int *cal)
 		return;
 	}
 
-	rc = property_get("ro.wcnss.serialno", serial, "");
-	if (rc) {
-		serial_num = convert_string_to_hex(serial);
-		ALOGE("Serial Number is  %x", serial_num);
-
-		msg[pos++] = WCNSS_USR_SERIAL_NUM >> BYTE_1;
-		msg[pos++] = WCNSS_USR_SERIAL_NUM >> BYTE_0;
-		msg[pos++] = serial_num >> BYTE_3;
-		msg[pos++] = serial_num >> BYTE_2;
-		msg[pos++] = serial_num >> BYTE_1;
-		msg[pos++] = serial_num >> BYTE_0;
-
-		if (write(fd, msg, pos) < 0) {
-			ALOGE("Failed to write to %s : %s", WCNSS_CTRL,
-					strerror(errno));
-			goto fail;
-		}
-	}
-
 #if defined(WCNSS_QMI) || defined (WCNSS_QMI_OSS)
 	if (SUCCESS == nv_mac_addr)
 	{
-- 
2.17.1

